src/rebar_prv_shell.erl

```diff
@@ -86,6 +89,7 @@ shell(State) ->
     setup_name(State),
     setup_paths(State),
     setup_shell(),
+    maybe_run_script(State),
```

Comment:
A gotcha there is that apps booting is what loads the `sys.config` file if any is defined. Do you expect the script to run after or before that has been loaded?

If it needs to run after, then it must run after the apps have been loaded, but before they've been booted, and should be moved down inside `maybe_boot_apps` (or rather, `maybe_boot_apps` needs to be blown up in: `maybe_load_apps`, `maybe_load_config`, `maybe_run_script` and then `maybe_boot_apps`) -- but that depends on the intent of the script.